pkg/compliance/checks/rego_check.go

```diff
@@ -244,16 +250,55 @@ func (r *regoCheck) appendInstance(input map[string][]interface{}, key string, i
 	input[key] = append(vars, instance.RegoInput())
 }
 
+func buildMappedInputs(inputs []compliance.RegoInput) map[string]compliance.RegoInput {
+	res := make(map[string]compliance.RegoInput)
+	for _, input := range inputs {
+		tagName := extractTagName(&input)
+
+		if _, present := res[tagName]; present {
+			log.Warnf("error building mapped input context: duplicated tag")
+			return nil
+		}
+
+		res[tagName] = input
+	}
+	return res
+}
+
+func roundTrip(inputs interface{}) (interface{}, error) {
+	output, err := yaml.Marshal(inputs)
+	if err != nil {
+		return nil, err
+	}
+	var res interface{}
+	if err := yaml.Unmarshal(output, &res); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
 func (r *regoCheck) buildContextInput(env env.Env) eval.RegoInputMap {
 	context := make(map[string]interface{})
+	context["id"] = r.ruleID
 	context["hostname"] = env.Hostname()
+
 	if r.ruleScope == compliance.KubernetesClusterScope {
 		context["kubernetes_cluster"], _ = env.KubeClient().ClusterID()
 	}
 	if r.ruleScope == compliance.KubernetesNodeScope {
 		context["kubernetes_node_labels"] = env.NodeLabels()
 	}
 
+	mappedInputs := buildMappedInputs(r.inputs)
+	if mappedInputs != nil {
+		preparedInputs, err := roundTrip(mappedInputs)
```

Comment:
Is it to ensure that we have only "basic" types ?

Reply:
The goal is to do a kind of `yaml` to `json conversion`.
The intended feature is for example to use the user provided file path, in the case where the file does not exist (so the usual input document doesn't have the field) to generate a user friendly error message

Reply:
I wanted to do a verbatim representation at first, but after talking with @lb5tr we think that it can be better to use a map with tags as keys